public_html/src/Demos/ConditionalDemo/ConditionalDemo.js

```diff
@@ -0,0 +1,599 @@
+        const rightVertical = this._createBorderRen(center[0] + (boxWidth * 0.5),
+                                                    center[1],
+                                                    renWidth,
+                                                    boxWidth * this.mNumBoxPerRow);
+        const topHorizontal = this._createBorderRen(center[0],
+                                                    center[1] + (boxWidth * 0.5),
+                                                    boxWidth * this.mNumBoxPerRow,
+                                                    renWidth);
+        const bottomHorizontal = this._createBorderRen(center[0],
+                                                       center[1] - (boxWidth * 0.5),
+                                                       boxWidth * this.mNumBoxPerRow,
+                                                       renWidth);
+                                                       
+        this.mMatrixRenderables = {top, bottom, left, right, leftVertical, rightVertical, topHorizontal, bottomHorizontal};
+        this.mParameters = new Parameters(center, boxWidth, renWidth);
+    }
+    
+    // Initialize the action's 3 * 3 double array
+    _initActionArray() {
+        this.mActionsArr = [];
+        this.mActionsRens = [];
+        for (var i = 0; i < 3; i++) {
+            this.mActionsArr.push([-1,-1,-1]);
+        }
+    }
+    
+    // Called from initialize. Initialize the mCamera
+    _initCamera() {
+        // Set up the cameras
+        this.mCamera = new Camera(
+          vec2.fromValues(100, 75), // position of the camera
+          200, // width of camera
+          [0, 0, 800, 600] // viewport (orgX, orgY, width, height)
+        );
+        this.mCamera.setBackgroundColor([0.8, 0.8, 0.8, 1]);
+        // sets the background to gray
+    }
+    
+    // Called from _initTurnSystem. Initialize the users in the game.
+    _initUsers() {
+        var player1 = new Player(0, "Player 1: Cross", [0,0,0,1], this.kFont, this.kCrossSprite, actionTypes.CROSS, 2, 15);
+        var player2 = new Player(1, "Player 2: Circle", [0,0,0,1], this.kFont, this.kCircleSprite, actionTypes.CIRCLE, 2, 10);
+        this.mTurnSystem.addUser(player1);
+        this.mTurnSystem.addUser(player2);
+    }
+    
+    // Called from initialize. Initialize the mTurnSystem
+    _initTurnSystem() {
+        // Define Conditional callbackfunction
+        function conditionFun (user) {
+            if (user.isMetCondition === true) {
+                return true;
+            }
+            return false;
+        }
+        
+        // use the builder class for setting
+        var settingBuilder = new Settings.Builder();
+        settingBuilder.setTurnType(this.mTurnType); // 'conditional'
+        settingBuilder.setMaxUsers(this.mMaxPlayerNum); // only 2 players 
+        settingBuilder.setCallbackFunction(conditionFun);
```

Comment:
Nit: but this could be:
```suggestion
        settingBuilder.setCallbackFunction((user) => users.isMetCondition);
```
I think that would give the same results